kafka/producer.go

```diff
@@ -84,28 +86,15 @@ func (p *Producer) produce(msg *Message, msgFlags int, deliveryChan chan Event)
 
 	crkt := p.handle.getRkt(*msg.TopicPartition.Topic)
 
-	var valp *byte
-	var keyp *byte
-	var empty byte
-	valLen := 0
-	keyLen := 0
-
-	if msg.Value != nil {
-		valLen = len(msg.Value)
-		// allow sending 0-length messages (as opposed to null messages)
```

Comment:
There is a difference between an empty Vallue/Key field (length 0) and a Null message (Value/Key is nil).
Empty messages are treated like any other message, just empty, while Null messages values are considered tombstones and null message keys are considered key-less.
That's the reason why the original code initialized both valp and keyp to nil (NULL in C) and I'm guessing this will break the cast-on-same-line requirement of Go.
One way around this might be to have four different code paths depending on the nullity of key and value, e.g.:
```
 if  key && val {
    produce(..., key, val, ..)
 else if key && !val {
   produce(..., key, C.NULL, ..)
 ..and so on
```